File: Java/RecursionPractice/src/calculations/FactorialTest.java
package calculations;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class FactorialTest {

    Factorial test1;

    Factorial test2;

    Factorial test3;

    @BeforeEach
    void setUp() {
        this.test1 = new Factorial(5);
        this.test2 = new Factorial(10);
        this.test3 = new Factorial(0);
    }

    @Test
    void getValue() {
        assertEquals(this.test1.getValue(), 120);
        assertEquals(this.test2.getValue(), 3628800);
        assertEquals(this.test3.getValue(), 1);
    }
}

File: Kotlin/RecursionPractice/test/calculations/FactorialTest.kt
package calculations

import org.junit.jupiter.api.Test

import org.junit.jupiter.api.Assertions.*

class FactorialTest {

    private val test1 = Factorial(5)

    private val test2 = Factorial(10)

    private val test3 = Factorial(0)

    @Test
    fun getValue() {
        assertEquals(test1.getValue(), 120)
        assertEquals(test2.getValue(), 3628800)
        assertEquals(test3.getValue(), 1)
    }
}